src/main/java/org/opentripplanner/api/common/RoutingResource.java

```diff
@@ -720,6 +721,19 @@ protected RoutingRequest buildRequest() throws ParameterException {
             }
         }
 
+
+         if (OTPFeature.DataOverlay.isOn() && otpServer.getRouter().davaOverlayConfig != null) {
+            request.genericGridDataRequestParam = GenericFileConfigurationParser.parseConfParam(this.otpServer.getRouter().davaOverlayConfig);
+             if (request.genericGridDataRequestParam != null) {
+                 for (Map.Entry<RequestParameters, RequestParameters> genParam : request.genericGridDataRequestParam.entrySet()) {
+                     RequestParameters thresholdEmpty = genParam.getKey();
+                     RequestParameters penaltyEmpty = genParam.getValue();
+                     thresholdEmpty.setValue(requestParameters.get(thresholdEmpty.getName()));
+                     penaltyEmpty.setValue(requestParameters.get(penaltyEmpty.getName()));
+                 }
+             }
+        }
+
```

Comment:
Is this initializing some default values in the request - or am I missing something here?

Reply:
Requires a second look / explanation